src/test/prompt.spec.ts

```diff
@@ -0,0 +1,117 @@
+import { expect } from "chai";
+import * as sinon from "sinon";
+import * as inquirer from "inquirer";
+
+import * as FirebaseError from "../error";
+import * as prompt from "../prompt";
+
+describe("prompt", () => {
+  let inquirerStub: sinon.SinonStub;
+  const PROMPT_RESPONSES = {
+    some: "fool",
+    with: "values",
+  };
+
+  beforeEach(() => {
+    // Stub inquirer to return a set of fake answers.
+    inquirerStub = sinon.stub(inquirer, "prompt").resolves(PROMPT_RESPONSES);
+  });
+
+  afterEach(() => {
+    sinon.restore();
+  });
+
+  describe("prompt", () => {
+    it("should error if questions are asked in nonInteractive environment", async () => {
+      const o = { nonInteractive: true };
+      const qs: prompt.Question[] = [{ name: "foo" }];
+
+      expect(prompt.prompt(o, qs)).to.be.rejectedWith(FirebaseError, /required.+non\-interactive/);
+    });
+
+    it("should utilize inquirer to prompt for the questions", async () => {
+      const qs: prompt.Question[] = [
+        {
+          name: "foo",
+          message: "this is a test",
+        },
+      ];
+
+      await prompt.prompt({}, qs);
+
+      expect(inquirerStub).calledOnceWithExactly(qs);
+    });
+
+    it("should add the new values to the options object", async () => {
+      const options = { hello: "world" };
+      const qs: prompt.Question[] = [
+        {
+          name: "foo",
+          message: "this is a test",
+        },
+      ];
+
+      await prompt.prompt(options, qs);
+
+      expect(options).to.deep.equal(Object.assign({ hello: "world" }, PROMPT_RESPONSES));
+    });
+  });
+
+  describe("promptOnce", () => {
+    it("should provide a name if one is not provided", async () => {
+      await prompt.promptOnce({ message: "foo" });
+
+      expect(inquirerStub).calledOnceWith([{ name: "question", message: "foo" }]);
+    });
+
+    it("should return the value for the given name", async () => {
+      const r = await prompt.promptOnce({ name: "some" });
+
+      expect(r).to.equal("fool");
+      expect(inquirerStub).calledOnce;
+    });
+  });
+
+  describe("convertLabeledListChoices", () => {
```

Comment:
For these choices, can we have `CHOICES` be something similar to what init would give so that it's more clear how this is used?

```
const CHOICES = [
  {
    name: "database",
    label: "Database: Deploy Firebase Realtime Database Rules",
    checked: false
  },
  ...
]
```

Reply:
I don't see any value in making these "similar to what init would give". Tests shouldn't reflect the code in that way. The test values here match the types that are required and have moderately sensible values.

That said, since I was moving some stuff around, I made one fo them a little different to make sure there was a longer string involved. But still, the value of them shouldn't matter too much... 🙂

Reply:
I frequently go to the tests to get granular context into what the function is doing and how it may be used in the code. Since this is solely used in the init portion to convert feature choices, I only see positives in making the test reflect that. That way, from one glance of the test, you know, ah, this is to convert features into choices - something I really had to dig to understand previously.

Reply:
I believe that glosses over a *huge* assumption: the reader came to the code itself (`convertLabeledListChoices`) and couldn't understand what it did by looking at the function signature and comments in the source. At that point, ideally, the context from which they came should be irrelevant - this code should be able to be understood by _just_ looking at the source file.

Could you take a look at the source for `convertLabeledListChoices` and tell me if there can be improvements made there, either in code, typing, or comments?

A disclaimer of sorts: this is a utility function used in exactly once place. A _very_ strong argument can be made to simply get rid of this function, but I decided not to do that in order to keep the scope of this specific change (typescript `prompt`, update `inquirer`) small. Maybe that's actually the best move here - if you think so, I'm happy to remove it, but it makes for a larger review.

Reply:
Sure - added comments in the src file itself.

Reply:
It's been removed.